jovian/utils/commit.py

```diff
@@ -125,23 +125,32 @@ def commit(message=None,
         log('Failed to detect Jupyter notebook or Python script. Skipping..', error=True)
         return
 
+    # Check for commit from jupyter extension
+    is_jupyter_extension = False
+    if 'jupyter_extension' in kwargs:
+        is_jupyter_extension = kwargs['jupyter_extension']
+
     # Attempt to save Jupyter notebook
-    if in_notebook():
+    if in_notebook() and not is_jupyter_extension:
         save_notebook()
         log('Attempting to save notebook..')
         sleep(1)
 
     # Extract notebook/script filename
     filename = _parse_filename(filename)
     if filename is None:
-        log(FILENAME_MSG)
+        log(FILENAME_MSG, error=True)
+        if is_jupyter_extension:
+            return False, ""
```

Comment:
Returning 1 value in certain places and 2 values in other places is likely to cause unexpected errors. Let's keep it simple and always return 2 values, irrespective of whether we're in the extension:

```
return False, "some error message" # for cases where the commit is not successful
return True, "username/title/v/9" # for cases where the commit is successful
```

Please update all return statements in `jovian.commit` to return 2 values. In case of the successful result, webapp URL can be added from the extension to create the full URL

Reply:
- Im using the check for extension as jupyter implicitly prints the return value along with logs.

- As for the second element of the tuple, since all errors and warnings are captured directly via stderr i have passed a empty string.

Reply:
Got it, in that case, we can simply return one value from commit:
- `False` if commit failed
- `username/title/v/8` if the commit was successful

Reply:
Any specific reason to get the webapp URL from the extension instead of directly from the return value? 
As that will require another python statement to be ran via kernel and wait for that result

Reply:
No major reason, it's just a bit odd to return the full URL as the return value of a function. Ideally it should be some structured information like `("username/title", version)` which the calling function can use to display the result in whatever way required. `username/title/v/version` seemed like a good midway compromise.

Adding another Python statement will not have a significant overhead, as the entire block of Python code constructed as `jvn_commit` will be sent to the kernel at once.

Reply:
In any case, I guess we can simply return the URL for now, as you suggested, and figure out later if we need to return a structure value.

Final implementation:
- return URL if the commit is successful
- return None (i.e. just `return`) if the commit is unsuccessful